src/backend/distributed/utils/citus_clauses.c

```diff
@@ -269,7 +269,18 @@ EvaluateNodeIfReferencesFunction(Node *expression, PlanState *planState)
 		}
 	}
 
-	return expression;
+	/* descend into expression tree (e.g. evaluate ARRAY[now(),now()]) */
```

Comment:
This is a little weird because `PartiallyEvaluateExpresionMutator` already recurses with the help of `expression_tree_mutator`. It's also a bit strange that `ExecuteMasterEvaluableFunctions` takes care in picking out which parts of the query should be mutated but here we're indiscriminately calling `query_tree_mutator`. I commented with a diff which also makes the test pass and seems more in line with how the code already works.